x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportExplainDataFrameAnalyticsAction.java

```diff
@@ -84,17 +100,37 @@ protected void doExecute(Task task,
 
     private void explain(Task task, PutDataFrameAnalyticsAction.Request request,
                          ActionListener<ExplainDataFrameAnalyticsAction.Response> listener) {
-        ExtractedFieldsDetectorFactory extractedFieldsDetectorFactory =
-                new ExtractedFieldsDetectorFactory(new ParentTaskAssigningClient(client, task.getParentTaskId()));
-        extractedFieldsDetectorFactory.createFromSource(
-            request.getConfig(),
-            ActionListener.wrap(
-                extractedFieldsDetector -> explain(task, request, extractedFieldsDetector, listener),
-                listener::onFailure)
+
+        final ExtractedFieldsDetectorFactory extractedFieldsDetectorFactory = new ExtractedFieldsDetectorFactory(
+            new ParentTaskAssigningClient(client, task.getParentTaskId())
         );
+        if (licenseState.isSecurityEnabled()) {
+            useSecondaryAuthIfAvailable(this.securityContext, () -> {
+                // Use secondary auth headers for the request always
```

Comment:
What does "always" mean in this context? We're in the conditional branch, that's why I'm asking.

Reply:
I think it means in preference to headers that might already be stored on the data frame analytics config, if it came from the config index rather than being supplied in the explain request.  The comment should say something like this.

So this means that if Alice tries to explain a pre-existing data frame analytics job created by Bob then that explanation will use Alice's credentials to access the source indices instead of Bob's.  Before this change it would have used Bob's, which is superficially analogous to what what would have happened if Alice had started the job that Bob created.  However, in the case of Alice starting Bob's job the outputs would go to the destination indices, which presumably Bob owns.  If Bob has chosen not to give access to Alice then Alice cannot see the results.  In the case of explain, Alice gets to see the results directly in the JSON response, so it makes more sense to use her credentials to access the source indices.